src/tools/ScenarioMeasurement/Startup/ProcessTimeParser.cs

```diff
@@ -70,9 +71,7 @@ public IList<Counter> Parse(string mergeTraceFile, string processName)
 
                 source.Kernel.ProcessEndGroup += evt =>
                 {
-                    // not using processname here because it's not tracked by the state machine when the process just ended
-                    // but ImageFileName exists
-                    if (pid.HasValue && evt.ProcessID == pid && evt.ImageFileName == imageFileName)
+                    if (pid.HasValue && imageFile.Equals(evt.ImageFileName, StringComparison.OrdinalIgnoreCase) && pid == evt.ProcessID)
```

Comment:
Because we're just using PID now we don't need to check the image file name here. It's not going to happen that the PID is reused while the process is still running.

Reply:
I guess it is still the same concern that Brian mentioned - there's a small chance the pid could be recycled in the same tracing session. I assume it means after one process ended, the next process would (for a very little chance) reuses the pid in the same tracing session. I'm not sure what you mean by "the process is still running".

Reply:
This is the end event for the process. It's incredibly unlikely to get a process end event for the PID we have stored for another instance of the process. For this to happen we would need:

1) We capture a PID and store it into `pid`
2) We miss the end event for whatever reason, and do not execute this code which clears `pid`
3) We miss the process start event for another instance of our test process **using the same PID**
4) We capture the end event for this second process.

In light of this I think we can simplify and not capture `imageFile`.

Reply:
I think that number three is not quite correct. It does not have to be our test process that starts, as any process that got assigned the recycled pid would trigger this ProcessEngGroup code, and we would not even have to miss the start event for this process, because it would not have the same name as our target process and we would never run that code.

Reply:
Sorry, I was describing the situation where `imageFile` saves us (but wasn't clear about that), which I think is very unlikely. I agree that some other process could use the PID and we'd get into a weird state if we miss 2. (I also think that's very very unlikely, particularly since we're using such low volume eventing for measurement.)